### PR TITLE
Include AWS Solution cloud formation template

### PR DESCRIPTION
**Pull request by peternied**
### Description
The existing AWS Solutions template is in a GitHub repo [GitHub - aws-solutions/migration-assistant-for-amazon-opensearch: Upgrade, Migrate, and Compare OpenSearch Clusters](https://github.com/aws-solutions/migration-assistant-for-amazon-opensearch) but isn’t used for performing updates.

With ongoing efforts to improve deployment speed, having the solutions deployment inside of the OpenSearch-Migration GitHub repo makes coordinating cross stack references easier and creates a clear workflow for contribution.

### Issues Resolved
- Open-source the AWS Solution CFN template source into the OpenSearch-Migration GitHub repository. https://opensearch.atlassian.net/browse/MIGRATIONS-2154

### To Do Before Leaving Draft
- [x] Update CI workflows
- [x] Update top level readme

### Check List
- [x] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: deployment/migration-assistant-solution/lib/solutions-stack.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        ]
+
+        const bootstrapRole = new Role(this, 'BootstrapRole', {
+            assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+            description: 'EC2 Bootstrap Role'
+        });
+        bootstrapRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AdministratorAccess'))
+
+        new InstanceProfile(this, 'BootstrapInstanceProfile', {
+            instanceProfileName: `bootstrap-${stageParameter.valueAsString}-instance-profile`,
```
Comment (lewijacn) [inline deployment/migration-assistant-solution/lib/solutions-stack.ts:131]: To resolve this bug: https://opensearch.atlassian.net/browse/MIGRATIONS-2027, can you also include region in this name

**Review by lewijacn — COMMENTED** [review]
I was expecting a bin/ folder do you plan to use?

File: deployment/migration-assistant-solution/lib/solutions-stack.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            blockDevices: [
+                {
+                    deviceName: "/dev/xvda",
+                    volume: BlockDeviceVolume.ebs(50)
+                }
+            ],
+            init: CloudFormationInit.fromElements(...cfnInitConfig),
+            initOptions: {
+                printLog: true,
+                ignoreFailures: true,
```
[the remark this answers is not in the document]
Comment (peternied) [reply deployment/migration-assistant-solution/lib/solutions-stack.ts:150]: Good catch, I'll removed this.  This was a hold over from when I was testing running the build as part of the EC2 startup.

File: deployment/migration-assistant-solution/lib/solutions-stack.ts (hunk 1 above)
Comment (peternied) [reply deployment/migration-assistant-solution/lib/solutions-stack.ts]: Done!

File: deployment/migration-assistant-solution/lib/solutions-stack.ts [hunk 3]
```diff
@@ -128,13 +128,13 @@ export class SolutionsInfrastructureStack extends Stack {
...
         new InstanceProfile(this, 'BootstrapInstanceProfile', {
-            instanceProfileName: `bootstrap-${stageParameter.valueAsString}-instance-profile`,
+            instanceProfileName: `bootstrap-instance-profile-${stackMarker}`,
             role: bootstrapRole
         })
 
         new Instance(this, 'BootstrapEC2Instance', {
             vpc: vpc,
-            instanceName: `bootstrap-${stageParameter.valueAsString}-instance`,
+            instanceName: `bootstrap-instance-${stackMarker}`,
```
Comment (lewijacn) [inline deployment/migration-assistant-solution/lib/solutions-stack.ts:137]: I'm not opposed to changing this naming pattern, but this definitely requires an update to our Solutions documentation which expect the previous naming pattern so just putting on your radar

**Review by lewijacn — APPROVED** [review]
See my comment for something to keep on your radar to update in our Solutions documentation

**Pull request merged**